### PR TITLE
JSON output flag for diff command

### PR DESCRIPTION
**Pull request by acunniffe**
## 🍗 Description
Two customers has asked for a `--json` output to help generate their company's public changelogs. 

This PR adds a `--json` flag that serializes the terminal changelog from `optic diff`. 

## 📚 References
_Links to relevant docs (Notion, Twist, GH issues, etc.), if applicable._

## 👹 QA
_How can other humans verify that this PR is correct?_

### CODE REVIEW

File: projects/openapi-utilities/src/utilities/json-changelog.ts [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+  return {
+    name: `${method} ${path}`,
+    change: getChange(change),
+    attributes: change
+      ? getDetailLogs(change, {
+          excludeKeys: ['pathPattern', 'method'],
+        })
+      : [],
+    parameters: parameterChanges,
+    requestBody:
```
Comment (acunniffe) [inline projects/openapi-utilities/src/utilities/json-changelog.ts:205]: if !added !change !removed -- is this always null?

Comment (niclim) [reply projects/openapi-utilities/src/utilities/json-changelog.ts:205]: Yeah it should be

File: projects/openapi-utilities/src/utilities/json-changelog.ts [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+  //   yield * indent(getParameterLogs('path', name, parameterChange));
+  // }
+  //
+  // for (const [name, parameterChange] of headers.changes) {
+  //   yield * indent(getParameterLogs('header', name, parameterChange));
+  // }
+  //
+  //
+  //
+  // yield '';
```
Comment (niclim) [inline projects/openapi-utilities/src/utilities/json-changelog.ts:245]: can we remove the dead code here?

File: projects/optic/src/commands/diff/diff.ts [hunk 3]
```diff
@@ -198,6 +188,21 @@ const runDiff = async (
     },
   };
 
+  if (specResults.changes.length === 0) {
+    console.log('No changes were detected');
+  } else {
+    console.log('');
+  }
+
+  if (options.json) {
```
Comment (niclim) [inline projects/optic/src/commands/diff/diff.ts:197]: If we're running this in JSON mode, should there be any other output? I presume we only want to dump a single json object out if `--json` is passed in?

Additionally, do you think it's valid to allow `--json` + `--web` opts to be specified at the same time?

**Review by niclim — APPROVED** [review]

**Pull request merged**